authors/apps/authentication/tests/test_views.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework.views import status
+
+from .base import BaseTestCase
+
+from ..models import User
+
+from .data import user_update, user1_info, login_info
+
+
+class RegistrationLoginViewTestCase(BaseTestCase):
+
+    def test_register_new_user_correct_details(self):
+        """Test api  can create user"""
+        
+        response = self.register_user(user1_info)
+
+        self.assertEqual(response.data['email'], "makabugo@andela.com")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_login_with_valid_credentials(self):
+        """Test api can login user if user password and email match"""
+
+        response = self.log_in_user(login_info)
+
+        self.assertEqual(response.data['username'], "fahad12")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class UserRetrieveUpdateViewTestCase(BaseTestCase):
+
+    def setUp(self):
+        self.user = User.objects.get(username="fahad12")
+        self.client.force_authenticate(user=self.user)
+        self.url = '/api/user/'
```

Comment:
@fahadmak Please use reverse from `django.urls` for this and anywhere else it appears